src/main/kotlin/me/partlysanestudios/partlysaneskies/features/gui/hud/rngdropbanner/RareDropGUI.kt

```diff
@@ -0,0 +1,197 @@
+
+class RareDropGUI : WindowScreen(ElementaVersion.V5) {
+    private val backgroundBox = UIBlock().constrain {
+        width = 75.percent
+        height = 75.percent
+        x = CenterConstraint()
+        y = CenterConstraint()
+        color = Color(0, 0, 0, 0).constraint
+    } childOf window
+
+    private val createFilterContainer = ThemeManager.currentBackgroundUIImage.constrain {
+        width = 60.percent
+        height = 75.percent
+        x = 40.percent
+        y = 0.percent
+    } childOf backgroundBox
+
+    private val createFiltersHeading = UIWrappedText("Create Filters:").constrain {
+        width = 90.percent
+        x = 5.percent
+        y = 2.percent
+        textScale = 1.5.scaledPixels
+        color = Color.gray.constraint
+    } childOf createFilterContainer
+
+    private val createFiltersInput = UITextInput("Create a ${currentFilterType.displayName} Filter...").constrain {
+        width = 90.percent
+        height = 5.percent
+        x = 5.percent
+        y = 7.percent
+    }.onMouseClick {
+        grabWindowFocus()
+    } as UITextInput childOf createFilterContainer
+
+    private val createFiltersScrollComponent = ScrollComponent(
+        scrollIconColor = primaryColor.toJavaColor(),
+        innerPadding = 10f,
+        scrollAcceleration = 2f,
+    ).constrain {
+        width = 100.percent
+        height = 90.percent
+        x = 0.percent
+        y = 10.percent
+    } childOf createFilterContainer
+
+    private val activeFiltersContainer = ThemeManager.currentBackgroundUIImage.constrain {
+        width = 35.percent
+        height = 100.percent
+        x = 0.percent
+        y = 0.percent
+    } childOf backgroundBox
+
+    private val activeFiltersHeading = UIWrappedText("${currentFilterType.displayName} Filters:").constrain {
+        width = 90.percent
+        x = 5.percent
+        y = 2.percent
+        textScale = 1.5.scaledPixels
+        color = Color.green.constraint
+    } childOf activeFiltersContainer
+
+    private val activeFiltersSearchBar: UITextInput = UITextInput("Search...").constrain {
+        width = 90.percent
+        height = 5.percent
+        x = 5.percent
+        y = 7.percent
+    }.onMouseClick {
+        grabWindowFocus()
+    }.onKeyType { _, _ ->
+        updateFilterList()
+    } as UITextInput childOf activeFiltersContainer
+
+    private val activeFiltersScrollComponent = ScrollComponent(
+        scrollIconColor = primaryColor.toJavaColor(),
+        innerPadding = 10f,
+    ).constrain {
+        width = 100.percent
+        height = 90.percent
+        x = 0.percent
+        y = 10.percent
+    } childOf activeFiltersContainer
+
+    private val presetsContainer = ThemeManager.currentBackgroundUIImage.constrain {
+        width = 60.percent
+        height = 20.percent
+        x = 40.percent
+        y = 80.percent
+    } childOf backgroundBox
+
+    init {
+        update()
+    }
+
+    fun update() {
+        updateFilterList()
+        addPresets()
```

Comment:
Pss buttons only work when you initiate them new every single time
Creating them at the top of the class doesn't work

Reply:
Huh?

Reply:
What doesn't work? I've never had an issue

Reply:
Like I alr mentioned in [dms 3 days](https://discord.com/channels/@me/1099173519062999172/1256609424873623592) ago, the button clicking didnt work there. This was because I created the PSS Button Instance at the top of the class
After moving that into the update() method, clicking the buttons worked for me.